test/translation.js

```diff
@@ -1,60 +1,160 @@
+    const requestHash = await translationInstance.getRequestHash(arrayOfIds[1]);
 
-    // const str = toAscii(result[1]).replace(/\0/g, "");
+    console.log("trying to get my string back. Hash is: " + requestHash)
+   
+    //convert to form consumable by IPFS
+    const bytes32toIPFS = bytes32ToIPFSHash(requestHash);
 
-    console.log("Translation String = ", str);
-    assert.equal(str, "Hi, how are you?");
+    //grab the request string from IPFS
+   await ipfs.cat(bytes32toIPFS, {buffer:true}).then(function(res){
+      console.log("hello there! " + res);
+      assert.equal(res.toString(), "Hi, how are you?");
+      
+    })
   });
 
   it("should return a failed transaction because the string has not been translated", async () => {
     const translationInstance = await TranslationContract.deployed();
-    const result = await translationInstance.getTranslatedString.call(0);
-    
-    console.log (result);
+    const badHash = await translationInstance.getTranslatedHash(0);
+    console.log(bytes32ToIPFSHash(badHash));
+    assert.equal(parseInt(badHash),0);
   });
 
   it("should return the translated string", async () => {
     const translationInstance = await TranslationContract.deployed();
 
-    await translationInstance.performTranslation("Bonjour, comment ça va?", 1);
-    const result = await translationInstance.getTranslatedString.call(1);
+    const performTranslation = async (obj) => {
+      //add translation to IPFS and existing Translation object
+      ipfs.add(new Buffer(obj.string), (err, result) => {
+          if (err) {
+              console.error("Content submission error:", err);
+              return false;
+          } else if (result && result[0] && result[0].hash) {
+              console.log("Successful translation. IPFS address:", result[0].hash);
+
+              //make contract calls to store the IPFS hash in Bytes32
+              translationInstance.performTranslation(
+                ipfsHashToBytes32(result[0].hash),
+                obj.translationID,
+              );  
 
-    console.log(result);
-    assert.equal(result,"Bonjour, comment ça va?");
+          } else {
+              console.log(result);
+              console.error("Unresolved content submission error");
+              return null;
+          }
+      });
+    }
+
+    const translation =  {
+      string: "Bonjour, comment ça va?", 
+      translationID: 1,
+    };
+    await performTranslation(translation);
+    
+    //remove this eventually - IPFS for some reason requires some delay in retrieving objects, so this dummy function call is here
+    const tmp = await translationInstance.getTmp(1);
+
+    const transHash = await translationInstance.getTranslatedHash(1);
+
+    const bytes32toIPFS = bytes32ToIPFSHash(transHash);
+    
+    console.log(transHash);
   
-  })
+    //retrieve translated string from IPFS
+    await ipfs.cat(bytes32toIPFS, {buffer:true}).then(function(res){
```

Comment:
You shouldn't need to use `.then` if you're using `async/await`.

Also, use arrow functions rather than the `function` literal.